sdk/storage/azure-storage-file-share/azure/storage/fileshare/_directory_client.py

```diff
@@ -115,7 +119,11 @@ def __init__(
         self._query_str, credential = self._format_query_string(
             sas_token, credential, share_snapshot=self.snapshot)
         super(ShareDirectoryClient, self).__init__(parsed_url, service='file-share', credential=credential, **kwargs)
-        self._client = AzureFileStorage(url=self.url, base_url=self.url, pipeline=self._pipeline)
+        self.allow_trailing_dot = kwargs.pop('allow_trailing_dot', None)
```

Comment:
I wonder if we should accept these augments on the higher-level clients (e.g. `ShareClient` and even the service client) so you can create a `ShareClient` with this option and then all file/directory clients you make from it will carry the setting through.